File: Competitive/RPC/2020/RPC02/feeding.java
import java.util.*;

public class feeding {

	public static void main(String[] args) {

		Scanner in = new Scanner(System.in);

		int n = in.nextInt();
		int c = in.nextInt();

		int[] vec = new int[n];

		for (int i = 0; i < n; i++) {
			vec[i] = in.nextInt();
		}
		Arrays.sort(vec);
		int x = 0, i = n - 1, j = 0;
		while (i > j) {
			if (vec[j] > (c - vec[i])) {
				x++;
				i--;
			} else {
				j++;
				i--;
				x++;
			}
		}
		if (i == j)
			x++;
		System.out.println(x);
	}
}

File: Competitive/RPC/2020/RPC02/feeding.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
  int n; int64_t m; cin>>n>>m;
  vector<int64_t> vec(n);
  for (auto &i: vec)
    cin>>i;
  sort(vec.begin(),vec.end());

  int x = 0, i = n - 1, j = 0;
  while (i > j) {
     if (vec[j] > (m - vec[i])) {
         x++;
         i--;
     } else {
         j++;
         i--;
         x++;
        }
    }
    if (i == j)
        x++;

  cout<<x<<endl;
}
